sources/osgShader/Compiler.js

```diff
@@ -742,25 +737,11 @@ define( [
             inputs = MACROUTILS.objectMix( inputs, shadowUniforms );
 
             // shadowTexture  Attribute uniforms AND varying
-            var tex;
             // TODO: better handle multi texture shadow (CSM/PSM/etc.)
             for ( k = 0; k < shadowTextures.length; k++ ) {
-
                 shadowTexture = shadowTextures[ k ];
                 if ( shadowTexture ) {
-                    tex = this.getOrCreateSampler( 'sampler2D', shadowTexture.getName() );
-                    inputs.shadowTexture = tex;
-
-                    // per texture uniforms
-                    var shadowTextureUniforms = this.getOrCreateTextureStateAttributeUniforms( shadowTexture, 'shadowTexture', k );
-                    inputs = MACROUTILS.objectMix( inputs, shadowTextureUniforms );
-
-                    var shadowVarying = {
-                        vertexWorld: vertexWorld,
-                        lightEyeDir: inputs.lightEyeDir,
-                        lightNDL: inputs.lightNDL
-                    };
-                    inputs = MACROUTILS.objectMix( inputs, shadowVarying );
+                    this.shadowTextureCreateUniformsInputAndVarying( shadowTexture, inputs, vertexWorld, k );
```

Comment:
use the same scheme of naming method something like
createShadowTextureAndVarying
createShadowTextureInput
maybe better